crates/katana/rpc/rpc-types/src/error/starknet.rs

```diff
@@ -83,6 +83,8 @@ pub enum StarknetApiError {
     TooManyKeysInFilter,
     #[error("Failed to fetch pending transactions")]
     FailedToFetchPendingTransactions,
+    #[error("Chunk Size is too big")]
+    ChunkSizeTooBig { requested: u64, maximum: u64 },
```

Comment:
we already have an error variant for this 

https://github.com/dojoengine/dojo/blob/0c70748fff67947895f3ff9255ef56ad986a5f41/crates/katana/rpc/rpc-types/src/error/starknet.rs#L34-L35

we can update the `PageSizeTooBig` variant into a struct variant and update the `.data()` function to include the struct fields,

https://github.com/dojoengine/dojo/blob/0c70748fff67947895f3ff9255ef56ad986a5f41/crates/katana/rpc/rpc-types/src/error/starknet.rs#L131-L145